examples/llm/scripts/export_for_inference.py

```diff
@@ -0,0 +1,180 @@
+
+    # Build Model
+    print('Initializing model...')
+    orig_model = build_composer_model(cfg.model)
+
+    # Loading checkpoint using Trainer
+    print('Loading model weights...')
+    trainer = Trainer(model=orig_model,
+                      load_path=load_path,
+                      load_weights_only=True)
+    # load export model with torch attention
+    attn_impl = cfg.model.get('attn_impl', 'torch')
+    if attn_impl == 'triton' or attn_impl == 'flash':
+        print(
+            f'Replacing {cfg.model.attn_impl} attention with torch causal attention'
+        )
+        cfg.model.attn_impl = 'torch'
+        export_model = build_composer_model(cfg.model)
+        trainer = Trainer(model=export_model,
+                          load_path=load_path,
+                          load_weights_only=True)
+        # replace flash/triton attention with torch causal attention
+        for idx in range(cfg.model.n_layers):
+            torch_causal_attn = TorchCausalAttention(cfg.model)
+            torch_causal_attn.mhsa.in_proj_weight = orig_model.model.transformer.blocks[
+                idx].causal_attn.mhsa.Wqkv.weight
+            torch_causal_attn.mhsa.in_proj_bias = orig_model.model.transformer.blocks[
+                idx].causal_attn.mhsa.Wqkv.bias
+            torch_causal_attn.mhsa.out_proj.weight = (
+                orig_model.model.transformer.blocks[idx].causal_attn.mhsa.
+                out_proj.weight)
+            torch_causal_attn.mhsa.out_proj.bias = orig_model.model.transformer.blocks[
+                idx].causal_attn.mhsa.out_proj.bias
+            export_model.model.transformer.blocks[
+                idx].causal_attn = torch_causal_attn
+    else:
+        export_model = orig_model
+
+    print('model loading done ...')
+
+    export_model.eval()
+
+    cpu_device = get_device('cpu')
+    cpu_device.module_to_device(export_model)
+
+    sample_input = gen_random_batch(export_batch_size, cfg)
+    # run forward pass on orig model once
+    with torch.no_grad():
+        export_model(sample_input)
+
+    save_path = cfg.get('export_save_path', 'model.onnx')
+    save_object_store = maybe_create_object_store_from_uri(save_path)
+    is_remote_store = save_object_store is not None
+    tempdir_ctx = (tempfile.TemporaryDirectory()
+                   if is_remote_store else contextlib.nullcontext(None))
+    with tempdir_ctx as tempdir:
+        if is_remote_store:
+            local_save_path = os.path.join(str(tempdir), 'model.onnx')
+        else:
+            local_save_path = save_path
+        trainer.export_for_inference(
+            save_format='onnx',
+            save_path=local_save_path,
+            sample_input=sample_input,
+        )
+
+        if cfg.get('verify_exported_model', False):
+            test_input = gen_random_batch(export_batch_size, cfg)
+            with torch.no_grad():
+                orig_out = export_model(test_input)
```

Comment:
Shouldn't this be:
```
orig_out = orig_model(test_input)
```

To cerify that replacing the flash attention with torch casual attention is math equivalent?

Reply:
We can't run the orig_model with flash/triton if this script runs on a cpu only system.

Reply:
Got it -- is it possible then to run this on a GPU system just to verify that the attention substitution process is indeed mathematically equivalent?

Reply:
Yes. Just verified. They don't match (bf16 for the GPU run while fp32 for the cpu run) with the tolerance we have but look close enough. 

tensor([ 0.0977,  0.4785,  0.2363,  ..., -0.3965,  0.1172,  0.1182],
       device='cuda:0', dtype=torch.bfloat16)
[ 0.09721886  0.47875217  0.23697849 ... -0.3966748   0.11686482
  0.11804749]